terragrunt/modules/aws-organization/groups.tf

```diff
@@ -112,6 +119,30 @@ locals {
         permissions : [aws_ssoadmin_permission_set.view_only_access] }
       ]
     },
+    # crates-io Staging
+    {
+      account : aws_organizations_account.crates_io_staging,
+      groups : [
+        { group : aws_identitystore_group.infra-admins,
+        permissions : [aws_ssoadmin_permission_set.view_only_access, aws_ssoadmin_permission_set.administrator_access] },
+        { group : aws_identitystore_group.infra,
+        permissions : [aws_ssoadmin_permission_set.view_only_access, aws_ssoadmin_permission_set.administrator_access] },
+        { group : aws_identitystore_group.crates_io,
+        permissions : [aws_ssoadmin_permission_set.view_only_access] },
+      ]
+    },
+    # crates-io Production
+    {
+      account : aws_organizations_account.crates_io_prod,
+      groups : [
+        { group : aws_identitystore_group.infra-admins,
+        permissions : [aws_ssoadmin_permission_set.view_only_access, aws_ssoadmin_permission_set.administrator_access] },
+        { group : aws_identitystore_group.infra,
+        permissions : [aws_ssoadmin_permission_set.view_only_access] },
+        { group : aws_identitystore_group.crates_io,
+        permissions : [aws_ssoadmin_permission_set.view_only_access] },
```

Comment:
This seems like an excellent starting point, but I expect us to require expansion as needs arise - view only access is basically metadata only, not enough to do regular actions (e.g., re-drive messages on the eventual queue).

Reply:
Agreed. I wasn't able to find prior art for this, and I don't think we actually discussed how exactly we want to deal with this. The separate accounts make it quite easy to grant pretty broad permissions, simplifying management. But I also feel like we need to make sure we have some oversight to prevent accidents or abuse. Let's discuss this further in #t-infra.